### PR TITLE
load connection string from ConnectionStrings section if default key not exist

### LINKED ISSUE #126: Consider adding `ConnectionStrings` in the fallback path when loading default connection string from environment
Azure Web App has a "Connection Strings" section specifically for secret connection strings. We should consider adding it into the fallback path when loading default connection string from environment, to optimize the experiences in Azure Web App.

### PR DESCRIPTION
**Pull request by xscript**
- fix #126

### CODE REVIEW

File: src/Microsoft.Azure.SignalR/HubHost/ServiceOptionsSetup.cs [hunk 1]
```diff
@@ -8,11 +8,22 @@ namespace Microsoft.Azure.SignalR
 {
     internal class ServiceOptionsSetup : IConfigureOptions<ServiceOptions>
     {
+        private static readonly string ConnectionStringSecondaryKey =
```
Comment (davidfowl) [inline src/Microsoft.Azure.SignalR/HubHost/ServiceOptionsSetup.cs]: Did you test it on azure web apps?

File: test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs [hunk 2]
```diff
@@ -85,8 +89,41 @@ public void AddAzureReadsConnectionStringFirst()
...
+        {
+            var services = new ServiceCollection();
+            var config = new ConfigurationBuilder()
+                            .AddInMemoryCollection(new Dictionary<string, string>
+                            {
+                                {"Azure:SignalR:ConnectionString", defaultValue},
+                                {"ConnectionStrings:Azure:SignalR:ConnectionString", secondaryValue}
+                            })
+                            .Build();
+            var serviceProvider = services.AddSignalR()
```
Comment (davidfowl) [inline test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs]: nit: This formatting looks a bit indented, can we reduce it a little

File: test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs [hunk 3]
```diff
@@ -85,8 +89,41 @@ public void AddAzureReadsConnectionStringFirst()
...
+        [InlineData(null, null, SecondaryValue, SecondaryValue)]
+        public void AddAzureSignalRLoadConnectionStringOrder(string customValue, string defaultValue,
+            string secondaryValue, string expected)
+        {
+            var services = new ServiceCollection();
+            var config = new ConfigurationBuilder()
+                            .AddInMemoryCollection(new Dictionary<string, string>
+                            {
+                                {"Azure:SignalR:ConnectionString", defaultValue},
+                                {"ConnectionStrings:Azure:SignalR:ConnectionString", secondaryValue}
```
Comment (davidfowl) [inline test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs]: This looks so strange 😄. `ConnectionStrings:Azure:SignalR:ConnectionString`

Should it be `ConnectionStrings:Azure:SignalR` instead?

Comment (chenkennt) [reply test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs]: Are ConnectionStrings section mainly for SQL databases? When I create a connection string in Azure portal, I need to choose from MySQL, SQLServer, SQLAzure, and Custom. What's the difference? Should user choose Custom when using SignalR?

Comment (davidfowl) [reply test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs]: Usually it's a database connection string. For example, I don't think you specify your service bus connection string in there.

File: src/Microsoft.Azure.SignalR/HubHost/ServiceOptionsSetup.cs (hunk 1 above)
Comment (xscript) [reply src/Microsoft.Azure.SignalR/HubHost/ServiceOptionsSetup.cs]: Yeah, tested. Works well.

File: test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs (hunk 2 above)
Comment (xscript) [reply test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs]: Sure.

File: test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs (hunk 3 above)
Comment (xscript) [reply test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs]: Agree that the key is repetitive and doesn't sound natural.
However this fix is specifically for Azure Web Apps. When user "accidentally" put the connection string in the `Connection Strings` section, we can still pick it up. So if we change the key name, it means we provide another key and need to document it.
IMO, this is not a recommended approach to set the connection string in environment. But if user prefer to put it in `Connection Strings` section, the Service SDK can still work.

File: test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs [hunk 4]
```diff
@@ -68,25 +69,55 @@ public void AddAzureReadsConnectionStringFirst()
...
+        [InlineData(null, null, SecondaryValue, SecondaryValue)]
+        public void AddAzureSignalRLoadConnectionStringOrder(string customValue, string defaultValue,
+            string secondaryValue, string expected)
+        {
+            var services = new ServiceCollection();
+            var config = new ConfigurationBuilder()
+                .AddInMemoryCollection(new Dictionary<string, string>
+                {
+                    {"Azure:SignalR:ConnectionString", defaultValue},
+                    {"ConnectionStrings:Azure:SignalR:ConnectionString", secondaryValue}
```
Comment (davidfowl) [inline test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs]: This key isn't correct right?

File: src/Microsoft.Azure.SignalR/HubHost/ServiceOptionsSetup.cs (hunk 1 above)
Comment (davidfowl) [reply src/Microsoft.Azure.SignalR/HubHost/ServiceOptionsSetup.cs]: We should change this to exclude the trailing ConnectionString right?

**Pull request merged**